plugins/out_parseable/parseable.c

```diff
@@ -0,0 +1,266 @@
+        else if (ctx->p_stream) {
+            /* Use the user-specified stream directly */
+            x_p_stream_value = flb_sds_create(ctx->p_stream);
+            if (!x_p_stream_value) {
+                flb_plg_error(ctx->ins, "Failed to set X-P-Stream header to the specified stream: %s", ctx->p_stream);
+                flb_sds_destroy(body);
+                msgpack_unpacked_destroy(&result);
+                FLB_OUTPUT_RETURN(FLB_ERROR);
+            }
+        }
+        else {
+            flb_plg_error(ctx->ins, "P_Stream is not set. Cannot determine the value for X-P-Stream.");
+            flb_sds_destroy(body);
+            msgpack_unpacked_destroy(&result);
+            FLB_OUTPUT_RETURN(FLB_ERROR);
+        }
+
+        /* Get upstream connection */
+        u_conn = flb_upstream_conn_get(ctx->upstream);
+        if (!u_conn) {
+            flb_plg_error(ctx->ins, "connection initialization error");
+            flb_sds_destroy(body);
+            flb_sds_destroy(x_p_stream_value);
+            msgpack_unpacked_destroy(&result);
+            FLB_OUTPUT_RETURN(FLB_ERROR);
+        }
+
+        /* Compose HTTP Client request */
+        client = flb_http_client(u_conn,
+                                 FLB_HTTP_POST, "/api/v1/ingest",
+                                 body, flb_sds_len(body),
+                                 ctx->p_server, ctx->p_port,
+                                 NULL, 0);
+
+        if (!client) {
+            flb_plg_error(ctx->ins, "could not create HTTP client");
+            flb_sds_destroy(body);
+            flb_sds_destroy(x_p_stream_value);
+            flb_upstream_conn_release(u_conn);
+            msgpack_unpacked_destroy(&result);
+            FLB_OUTPUT_RETURN(FLB_ERROR);
+        }
+
+        /* Add HTTP headers */
+        flb_http_add_header(client, "Content-Type", 12, "application/json", 16);
+        flb_http_add_header(client, "X-P-Stream", 10, x_p_stream_value, flb_sds_len(x_p_stream_value));
+        flb_http_basic_auth(client, ctx->p_username, ctx->p_password);
+
+        /* Perform request */
+        ret = flb_http_do(client, &b_sent);
+        flb_plg_info(ctx->ins, "HTTP request http_do=%i, HTTP Status: %i",
+                     ret, client->resp.status);
+
+        /* Clean up resources */
+        flb_sds_destroy(body);
+        flb_sds_destroy(x_p_stream_value);
+        flb_http_client_destroy(client);
+        flb_upstream_conn_release(u_conn);
+    }
+    msgpack_unpacked_destroy(&result);
+
+    FLB_OUTPUT_RETURN(FLB_OK);
+}
+
+
+static int cb_parseable_exit(void *data, struct flb_config *config)
+{
+    struct flb_out_parseable *ctx = data;
+
+    if (!ctx) {
+        return 0;
+    }
+
+    /* Free up resources */
+    if (ctx->upstream) {
+        flb_upstream_destroy(ctx->upstream);
+    }
+    flb_free(ctx);
+    return 0;
+}
+
+/* Configuration properties map */
+static struct flb_config_map config_map[] = {
+    {
+     FLB_CONFIG_MAP_STR, "P_Server", NULL,
```

Comment:
I'd say we should use the standard options we already have for http output. I'm guessing the P_ prefix is for parseable but as it's already in the config for that plugin it seems redundant so I'd suggest removing and aligning with the usual names

Reply:
we have standardised our configs, usual names as suggested.